test/tests/functional/pbs_qmgr.py

```diff
@@ -160,24 +160,42 @@ def test_multi_attributes(self):
             self.assertTrue(msg in e.msg[0])
         self.server.expect(NODE, 'queue', op=UNSET, id=self.mom.shortname)
 
-    def test_string_quoting(self):
+    def set_and_test_comment(self, comment):
         """
-        Test to verify that if a string attribute has spaces, the value
-        is quoted when printed
+        Set the node's comment, then print it and re-import it
         """
-        a = {'default_qsub_arguments': '-V -lplace=scatter'}
-        self.server.manager(MGR_CMD_SET, SERVER, a)
-        qmgr_cmd = \
-            os.path.join(self.server.pbs_conf['PBS_EXEC'], 'bin', 'qmgr') + \
-            " -c '%s'"
-        qmgr_cmd_print = qmgr_cmd % ('p s default_qsub_arguments')
+        a = {'comment': comment}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.hostname)
+        qmgr_path = \
+            os.path.join(self.server.pbs_conf['PBS_EXEC'], 'bin', 'qmgr')
+        qmgr_cmd_print = qmgr_path + \
+            (' -c "p n %s comment"' % self.mom.hostname)
         ret = self.du.run_cmd(self.server.hostname,
                               cmd=qmgr_cmd_print, as_script=True)
```

Comment:
qmgr output could be redirected to a file directly via run_cmd() 's stdout parameter. Other tests in the suite seem to do that. Any reason you want to do it this way ?

Reply:
For a reason unknown to me, when we run `print node <nodename> comment`, it prints the command to create the node and the command to set the comment. If we pipe it back into qmgr, it will print an error.

Reply:
The print option is meant to recreate.  The idea is you do a qmgr -c 'print server' > file and then on another server you redirect that back in to qmgr to recreate your settings.  So you are printing a node, you want to recreate that node.  This is why create node is printed.  If you wanted to list the attributes, you would use the list command.

Reply:
@vstumpf . Thanks for clarifying. I am fine with this way then.

Reply:
It would make more sense if I could print multiple attributes at once, for example `print node <nodename> resv_enable sharing vn_type`, and it would be able to recreate the node with the specific attributes. But because I can only print one at a time, I would have to print each separately and remove the duplicate `creates`.

But I understand probably no one uses it this way, so it's fine as it is.